core/controllers/admin.py

```diff
@@ -71,6 +76,108 @@ class AdminHandler(base.BaseHandler):
     """Handler for the admin page."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'POST': {
+            'action': {
+                'type': 'string',
+                'choices': [
+                    'reload_exploration', 'reload_collection',
+                    'generate_dummy_explorations', 'clear_search_index',
+                    'generate_dummy_new_structures_data',
+                    'generate_dummy_new_skill_data', 'save_config_properties',
+                    'revert_config_property', 'upload_topic_similarities',
+                    'start_computation', 'stop_computation',
+                    'cancel_job', 'regenerate_topic_related_opportunities',
+                    'start_new_job', 'regenerate_missing_exploration_stats',
+                    'update_feature_flag_rules'
+                ],
+                'default_value': None
+            },
+            'exploration_id': {
+                'type': 'string',
+                'default_value': None
+            },
+            'collection_id': {
+                'type': 'string',
+                'default_value': None
+            },
+            'num_dummy_exps_to_generate': {
+                'type': 'int',
+                'default_value': None
+            },
+            'num_dummy_exps_to_publish': {
+                'type': 'int',
+                'default_value': None
+            },
+            'new_config_property_values': {
+                'type': 'object_dict',
+                'validation_method': (
+                    validation_method.validate_new_config_property_values),
+                'default_value': None
+            },
+            'config_property_id': {
+                'type': 'string',
+                'default_value': None
+            },
+            'job_type': {
+                'type': 'string',
+                'default_value': None
+            },
+            'job_id': {
+                'type': 'string',
+                'default_value': None
+            },
+            'computation_type': {
+                'type': 'string',
+                'default_value': None
+            },
+            'data': {
+                'type': 'string',
+                'default_value': None
+            },
+            'topic_id': {
+                'type': 'string',
+                'default_value': None
+            },
+            'feature_name': {
```

Comment:
feature_name is restricted so I think you can add choices in the schema with ALL_FEATURES_NAMES_SET

Reply:
Done